src/ziggurat/producer.clj

```diff
@@ -55,27 +57,79 @@
    :methods  [^{:static true} [send [String String Object Object] java.util.concurrent.Future]
               ^{:static true} [send [String String int Object Object] java.util.concurrent.Future]]))
 
-(defn- producer-properties-from-config [{:keys [bootstrap-servers
-                                                acks
-                                                key-serializer
-                                                value-serializer
-                                                enable-idempotence
-                                                retries-config
-                                                max-in-flight-requests-per-connection]}]
-  (doto (Properties.)
-    (.put ProducerConfig/BOOTSTRAP_SERVERS_CONFIG bootstrap-servers)
-    (.put ProducerConfig/ACKS_CONFIG acks)
-    (.put ProducerConfig/RETRIES_CONFIG (int retries-config))
-    (.put ProducerConfig/ENABLE_IDEMPOTENCE_CONFIG enable-idempotence)
-    (.put ProducerConfig/MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION (int max-in-flight-requests-per-connection))
-    (.put ProducerConfig/KEY_SERIALIZER_CLASS_CONFIG key-serializer)
-    (.put ProducerConfig/VALUE_SERIALIZER_CLASS_CONFIG value-serializer)))
+(defn *implements-serializer?* [serializer-class]
+  (contains? (set (.getInterfaces (Class/forName serializer-class)))
+             (Class/forName "org.apache.kafka.common.serialization.Serializer")))
+
+(def implements-serializer? (s/pred *implements-serializer?* 'implements-serializer?))
+
+(s/defschema ProducerConfigSchema {(s/required-key :bootstrap-servers)                     s/Any
+                                   (s/optional-key :key-serializer-class)                  implements-serializer?
+                                   (s/optional-key :value-serializer-class)                implements-serializer?
+                                   (s/optional-key :key-serializer)                        implements-serializer?
+                                   (s/optional-key :value-serializer)                      implements-serializer?
+                                   (s/optional-key :retries-config)                        s/Any
```

Comment:
don't use s/any. put specific validators

Reply:
Hey @kartik7153, This PR is already delayed a lot. We can address specific validations as refactorings later on.

Reply:
This is not refactoring. This will break if you pass the wrong data types to java client since you are doing s/any.

Reply:
It won't. Everything is parsed to string here. https://github.com/murtaza0xFF/ziggurat/blob/0c56aa6a2681b6db79dbc1d4c9d057c654477fe8/src/ziggurat/producer.clj#L124

Reply:
What about java client. Everything is str?

Reply:
Yes. https://kafka.apache.org/23/javadoc/org/apache/kafka/clients/producer/ProducerConfig.html

Reply:
https://kafka.apache.org/090/javadoc/index.html?org/apache/kafka/clients/producer/KafkaProducer.html

I don't everything is str. Let me know if I am wrong.

Reply:
@kartik7153 yeah I think it makes sense to just validate against the types defined here instead of str'ing everything. Let me know what you think. https://kafka.apache.org/documentation/#producerconfigs

Reply:
As discussed on Slack, I just verified and it seems like the current implementation is correct. Java's Property API requires the key and the value to be set to string. The properties are coerced to the correct types by KafkaProducer internally. The types on the documentation are an indication of what it's parsed to internally, as opposed to the types it expects.

[Here](https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java#L398) is the internals of KafkaProducer where it parses to the defined type.

That function is defined [here](https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java#L172 ). Notice that string is expected by every function that does the parsing.